legate/core/launcher.py

```diff
@@ -661,28 +819,33 @@ def add_unbound_output(self, store, fspace, field_id):
             )
         )
 
-    def add_future(self, future):
+    def add_future(self, future: Future) -> None:
         self._future_args.append(future)
 
-    def add_future_map(self, future_map):
+    def add_future_map(self, future_map: FutureMap) -> None:
         self._future_map_args.append(future_map)
 
-    def add_communicator(self, handle):
+    def add_communicator(self, handle: FutureMap) -> None:
         self._comms.append(handle)
 
-    def set_sharding_space(self, space):
+    def set_sharding_space(self, space: IndexSpace) -> None:
         self._sharding_space = space
 
-    def set_point(self, point):
+    def set_point(self, point: Point) -> None:
         self._point = point
 
     @staticmethod
-    def pack_args(argbuf, args):
+    def pack_args(
+        argbuf: BufferBuilder,
+        args: Sequence[Union[ScalarArg, RegionFieldArg, FutureStoreArg]],
```

Comment:
So the protocol should be for all these argument types .

Reply:
https://github.com/nv-legate/legate.core/issues/192